source/unified-test-format/tests/valid-pass/expectedEventsForClient-topologyDescriptionChangedEvent.yml

```diff
@@ -0,0 +1,41 @@
+description: "expectedEventsForClient-topologyDescriptionChangedEvent"
+
+schemaVersion: "1.20"
+
+runOnRequirements:
+  - topologies:
+      - replicaset
+    minServerVersion: "4.4" # awaitable hello
+
+tests:
+  - description: "can assert on values of newDescription and previousDescription fields"
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+            - client:
+                id: &client client
+                observeEvents:
+                  - topologyDescriptionChangedEvent
+      # ensure the topology has been fully discovered before closing the client.
+      # expected events are initial server discovery and 3 server connect events.
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            topologyDescriptionChangedEvent: {}
+          count: 1
+    expectEvents:
+      - client: *client
+        eventType: sdam
+        ignoreExtraEvents: true 
+        events:
+          - topologyOpeningEvent: {}
+          - topologyDescriptionChangedEvent: 
+              previousDescription:
+                type: "Unknown"
+              newDescription:
+                type: "ReplicaSetWithPrimary"
```

Comment:
When I run this test in Python, this first event is ReplicaSetNoPrimary, not ReplicaSetWithPrimary:
```python
<TopologyDescriptionChangedEvent topology_id: 66104eba924530e675e287ae changed from:
  <TopologyDescription id: 66104eba924530e675e287ae, topology_type: Unknown, servers: []>, to:
  <TopologyDescription id: 66104eba924530e675e287ae, topology_type: ReplicaSetNoPrimary, servers: [<ServerDescription ('localhost', 27019) server_type: Unknown, rtt: None>]>>
```

We do this when creating the client because in this case we know ahead of time we are connecting to a replica set (because our test suite adds the replicaSet option.

Reply:
How about we add `"directConnection": true` to the client and use `type: Single`?